src/std_impls.rs

```diff
@@ -0,0 +1,161 @@
+// Bitcoin Hashes Library
+// Written in 2019 by
+//   Andrew Poelstra <apoelstra@wpsoftware.net>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! `std` Impls
+//!
+//! impls of traits defined in `std` and not `core`
+
+use std::{error, io};
+
+use {sha1, sha256, sha512, ripemd160, siphash24};
+use HashEngine;
+use Error;
+
+impl error::Error for Error {
+    fn cause(&self) -> Option<&error::Error> { None }
+    fn description(&self) -> &str { "`std::error::description` is deprecated" }
+}
+
+impl io::Write for sha1::HashEngine {
```

Comment:
That's a lot of identical code. Would a macro be appropriate or are we trying to get rid of them?

Reply:
Heh, I couldn't decide. It's only 40 or 50 lines.